lib/GitCommit.php

```diff
@@ -2,21 +2,90 @@
 use Safe\DateTimeImmutable;
 
 class GitCommit{
+	public ?int $GitCommitId = null;
+	public ?int $EbookId = null;
 	public DateTimeImmutable $Created;
 	public string $Message;
 	public string $Hash;
 
 	/**
 	 * @throws Exceptions\InvalidGitCommitException
 	 */
-	public function __construct(string $unixTimestamp, string $hash, string $message){
+	public static function FromLog(string $unixTimestamp, string $hash, string $message): GitCommit{
```

Comment:
I suggest we refactor this to accept a string, since we're saying "create this object from a log line" and not "create this object from several structured parameters". The latter is more akin to `__construct`. Then this function would parse the log line string and create the object. Or, rename this to `__construct`.

Reply:
Done.